cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AppSystemMetadataWriter.java

```diff
@@ -95,13 +95,21 @@ private void addAccelerators(ApplicationSpecification appSpec,
       .flatMap(Set::stream)
       .collect(Collectors.toSet());
     //add from application
-    acceleratorSet.addAll(applicationClass.getRequirements().getAccelerators());
+    acceleratorSet.addAll(getApplicationAccelerators(applicationClass));
     if (acceleratorSet.isEmpty()) {
       return;
     }
     properties.put(ACCELERATOR_TAG, String.join(ACCELERATOR_DELIMITER, acceleratorSet));
   }
 
+  private Set<String> getApplicationAccelerators(ApplicationClass applicationClass) {
+    if (applicationClass == null || applicationClass.getRequirements() == null || applicationClass.getRequirements()
```

Comment:
Is `applicationClass` nullable?

Reply:
I am not sure if I am propogating it correctly. Need to debug further to check what is null here. This is to unblock.

Reply:
I don't think it can be `null`. I think it's only the requirements is null.

Reply:
N.A with latest changes